src/java.base/unix/native/libnio/fs/UnixNativeDispatcher.c

```diff
@@ -71,6 +73,95 @@
 #define readdir64 readdir
 #endif
 
+#if defined(__linux__)
+// Account for the case where we compile on a system without statx
+// support. We still want to ensure we can call statx at runtime
+// if the runtime glibc version supports it (>= 2.28)
+#ifndef __NR_statx
```

Comment:
As long as we have to accommodate the case where struct statx is not defined, we may just as well do it always: just define a structure binary compatible to statx but named differently (e.g. my_statx) to prevent name clashes in case the structure happens to be defined already from libc; then use that structure as return type for your function pointer.

The advantage would be you don't need to know and depend on __NR_statx, and you don't need to include asm/unistd. And no two cases to build and test.

We do this already in other places, see e.g. the dynamic inclusion of mallinfo2() in os_linux.cpp.

Reply:
OK.

Reply:
Should be done now in the latest patch.